File: JavaScript_Courses/JavaScript_Basics/Preparation/06_Nested_Loops_Exercise/05_Special_Numbers.js
function specialNumbers(input) {
    let num = Number(input[0]);

    let result = '';

    for (let currentNum = 1111; currentNum <= 9999; currentNum++) {
        let currentNumAsString = String(currentNum);
        let isValid = true;

        for (i = 0; i < currentNumAsString.length; i++) {
            let currentDigit = Number(currentNumAsString[i]);

            if (num % currentDigit != 0) {
                isValid = false;
                break;
            }
        }

        if (isValid) {
            result += `${currentNumAsString} `;
        }
    }

    console.log(result);
}


// specialNumbers(["3"]);

File: Python_Courses/Python_Basics/Python_PB_2023_04/06_Nested_Loops_Exercise/05_Special_Numbers.py
number = int(input())

for num in range(1_111, 10_000):  # num holds each of [1111, 1112, ..., 9999]

    is_special = True
    num_as_str = str(num)  # num_as_str "1111"
    for _, digit in enumerate(num_as_str):  # digit holds each of ["1", "1", "1", "1"]
        current_digit = int(digit)

        if current_digit == 0:
            is_special = False
            break

        if not number % current_digit == 0:
            is_special = False
            break

    if is_special:
        print(f"{num}", end=" ")
